content/en/docs/usage/make_build.md

```diff
@@ -0,0 +1,434 @@
+config APPHELLOWORLD_DEPENDENCIES
+        bool
+        default y
+        select LIBMUSL
+        select LIBLWIP
+
+config APPHELLOWORLD_LOGGER
+        bool "Add the logger to the build system"
+        default n
+```
+
+This will select the `musl` and `lwip` dependencies and create a new configuration option that can be set in the `Application Options` screen in the `menuconfig`.
+
+{{< img
+      class="w-auto mx-auto"
+      src="/assets/imgs/new-app-option.png"
+      position="center"
+>}}
+
+For more detailed information about using and configuring the build system, refer to the [porting page](docs/develop/porting/).
+
+## Example: Using Make with `app-helloworld-cpp`
+
+Following the [steps above](docs/usage/make_build/#building-app-helloworld-with-make), we will set up, configure, build and run the [`app-helloworld-cpp` application](https://github.com/unikraft/app-helloworld-cpp).
+
+To create the setup, we will start with the following directory structure:
+
+```text
+my-unikernel/
+	|-- apps/
+	|   `-- app-helloworld-cpp      <- helloworld-cpp application
+	|-- libs/                       <- additional libraries go here
+	`-- unikraft/                   <- Unikraft core
+```
+
+We can do this by executing the following sequence of commands:
+
+```console
+$ mkdir my-unikernel
+$ cd my-unikernel
+$ git clone https://github.com/unikraft/unikraft
+$ mkdir libs
+$ cd libs
+$ git clone https://github.com/unikraft/lib-musl musl
+$ git clone https://github.com/unikraft/lib-libunwind libunwind
+$ git clone https://github.com/unikraft/lib-libcxx libcxx
+$ git clone https://github.com/unikraft/lib-libcxxabi libcxxabi
+$ git clone https://github.com/unikraft/lib-compiler-rt compiler-rt
+$ mkdir apps
+$ cd apps
+$ git clone https://github.com/unikraft/app-helloworld-cpp
+$ cd app-helloworld-cpp
+```
+
+We can see that there is no `Makefile`, so we create one and add the external dependencies:
+
+```Makefile
+UK_ROOT ?= $(PWD)/../../unikraft
+UK_LIBS ?= $(PWD)/../../libs
+LIBS := $(UK_LIBS)/musl:$(UK_LIBS)/libunwind:$(UK_LIBS)/libcxx:$(UK_LIBS)/libcxxabi:$(UK_LIBS)/compiler-rt
+
+all:
+	@$(MAKE) -C $(UK_ROOT) A=$(PWD) L=$(LIBS)
+
+$(MAKECMDGOALS):
+	@$(MAKE) -C $(UK_ROOT) A=$(PWD) L=$(LIBS) $(MAKECMDGOALS)
+```
+
+There is already a `Makefile.uk` file present in the repository, containing the rule that adds the application to the build system and the source files.
+
+```Makefile
+$(eval $(call addlib,apphelloworldcpp))
+
+APPHELLOWORLDCPP_SRCS-y += $(APPHELLOWORLDCPP_BASE)/helloworld.cpp
+```
+
+After that, we configure the application:
+To do that, we can create a `Config.uk` file that contains the application dependencies:
+
+```kconfig
+config APPHELLOWORLD_CPP_DEPENDENCIES
+	bool
+	default y
+	select PLAT_KVM
+	select LIBMUSL
+	select LIBCXX
+	select LIBCOMPILER_RT
+```
+
+Then we can run `make menuconfig`, and just save the configuration.
+We can also run `make menuconfig` and manually select the required dependencies.
```

Comment:
```suggestion
Then we can run `make menuconfig`, and just save the configuration.
Nothing else is needed, because required options are loaded from the `Config.uk` file.

In case there were no `Config.uk` file, we could still create the configuration, but manually, by running `make menuconfig` and manually selectin the required options.
```